cmd/sidecars/kafka.go

```diff
@@ -13,25 +15,21 @@ type KafkaConfig struct {
 	AutoOffsetReset  string
 	SidecarImage     string
 	// insprdPort is the port used in the insprd service of your cluster
-	KafkaInsprPort string
+	KafkaInsprAddr string
 }
 
 // KafkaToDeployment receives a the KafkaConfig variable as a parameter and returns a
 // SidecarFactory function that is used to subscribe to the sidecarFactory
 func KafkaToDeployment(config KafkaConfig) models.SidecarFactory {
-	return func(app *meta.App, conn *models.SidecarConnections) k8s.DeploymentOption {
-		return k8s.WithContainer(
-			k8s.NewContainer(
-				"sidecar-kafka-"+app.Meta.UUID, // deployment name
-				config.SidecarImage,            // image url
+	return func(app *meta.App, conn *models.SidecarConnections, opts ...k8s.ContainerOption) (corev1.Container, []corev1.EnvVar) {
+		envVars, kafkAddr := KafkaSidecarConfig(config, conn)
 
-				// label to the dApp associated with it
-				InsprAppIDConfig(app),
-				KafkaEnvConfig(config),
-				KafkaSidecarConfig(config, conn),
-				k8s.ContainerWithPullPolicy(corev1.PullAlways),
-			),
-		)
+		return k8s.NewContainer(
+			"sidecar-kafka-"+app.Meta.UUID, // deployment name
+			config.SidecarImage,            // image url
+			// label to the dApp associated with it
```

Comment:
this comment can be removed, since the function bellow does more than what is being described